CMakeLists.txt

```diff
@@ -46,8 +46,12 @@ if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
     endif()
 endif()
 
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Werror=return-type")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Werror=return-type")
+if(MSVC)
+	add_definitions(-DNOMINMAX)
```

Comment:
maybe MSVC needs some warning flags as well? I'm not familiar enough with MSVC to judge that.

Reply:
There are several changes that needs to happen yeah - I intend to do a pass on CMake config for msvc later on.